src/extension/module.js

```diff
@@ -888,14 +888,25 @@
    */
   function fireEvent(sk, name, data) {
     try {
-      sk.dispatchEvent(new CustomEvent(name, {
-        detail: {
-          data: data || {
-            config: JSON.parse(JSON.stringify(sk.config)),
-            location: sk.location,
-            status: sk.status,
-          },
+      const { config, location, status } = sk;
+      data = data || {
+        // turn complex into simple objects for event listener
+        config: JSON.parse(JSON.stringify(config)),
+        location: {
```

Comment:
or: json clone:

```
JSON.parse(JSON.stringify(window.location))
```

Reply:
It's not `window.location`, but a `URL` object, which won't let itself stringify.